marconi/app/Main.hs

```diff
@@ -218,23 +246,25 @@ utxoWorker
   :: Coordinator
   -> TChan (ChainSyncEvent (BlockInMode CardanoMode))
   -> FilePath
+  -> Maybe FilteredAddresses
   -> IO ()
-utxoWorker Coordinator{_barrier} ch path = Utxo.open path (Utxo.Depth 2160) >>= innerLoop
+utxoWorker Coordinator{_barrier} ch path addresses= Utxo.open path (Utxo.Depth 2160) >>= innerLoop
   where
     innerLoop :: UtxoIndex -> IO ()
     innerLoop index = do
       signalQSemN _barrier 1
       event <- atomically $ readTChan ch
       case event of
         RollForward (BlockInMode (Block (BlockHeader slotNo _ _) txs) _) _ct ->
-          Ix.insert (getUtxoUpdate slotNo txs) index >>= innerLoop
+            Ix.insert (getUtxoUpdate slotNo txs addresses) index >>= innerLoop
```

Comment:
@koslambrou  [PLT-599](https://input-output.atlassian.net/browse/PLT-599) specifically mentions filtering for utxo address, so I scoped the changes to this worker.

Reply:
Yep! That's what I expected :)

Reply:
thnx @koslambrou .I'll add the corresponding unit tests and will put it for review